controllers/zookeeperbackup_controller.go

```diff
@@ -0,0 +1,403 @@
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"net/http"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"strconv"
+	"strings"
+
+	"github.com/mitchellh/hashstructure/v2"
+)
+
+// ReconcileTime is the delay between reconciliations
+const PVCSuffix = "-pvc"
+
+var logBk = logf.Log.WithName("controller_zookeeperbackup")
+var hash uint64
+
+type LeaderGetter func(hostname string, port int32) (string, error)
+
+// ZookeeperBackupReconciler reconciles a ZookeeperBackup object
+type ZookeeperBackupReconciler struct {
+	Client       client.Client
+	Scheme       *runtime.Scheme
+	Log          logr.Logger
+	LeaderGetter LeaderGetter
+}
+
+//+kubebuilder:rbac:groups=zookeeper.pravega.io.zookeeper.pravega.io,resources=zookeeperbackups,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=zookeeper.pravega.io.zookeeper.pravega.io,resources=zookeeperbackups/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=zookeeper.pravega.io.zookeeper.pravega.io,resources=zookeeperbackups/finalizers,verbs=update
+
+func (r *ZookeeperBackupReconciler) Reconcile(_ context.Context, request reconcile.Request) (reconcile.Result, error) {
+	r.Log = logBk.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	r.Log.Info("Reconciling ZookeeperBackup")
+
+	// Fetch the ZookeeperBackup instance
+	zookeeperBackup := &zookeeperv1beta1.ZookeeperBackup{}
+	err := r.Client.Get(context.TODO(), request.NamespacedName, zookeeperBackup)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+	changed := zookeeperBackup.WithDefaults()
+	if changed {
+		r.Log.Info("Setting default settings for zookeeper-backup")
+		if err := r.Client.Update(context.TODO(), zookeeperBackup); err != nil {
+			return reconcile.Result{}, err
+		}
+		return reconcile.Result{Requeue: true}, nil
+	}
+
+	// Check if zookeeper cluster exists
+	foundZookeeperCluster := &zookeeperv1beta1.ZookeeperCluster{}
+	zkCluster := zookeeperBackup.Spec.ZookeeperCluster
+	err = r.Client.Get(context.TODO(), types.NamespacedName{Name: zkCluster, Namespace: zookeeperBackup.Namespace}, foundZookeeperCluster)
+	if err != nil && errors.IsNotFound(err) {
+		r.Log.Error(err, fmt.Sprintf("Zookeeper cluster '%s' not found", zkCluster))
```

Comment:
Please, log ZookeeperBackup name and the namespace.

Reply:
The same.